Source/Urho3D/Engine/Engine.cpp

```diff
@@ -1061,11 +1074,130 @@ void Engine::DefineParameters(CLI::App& commandLine, StringVariantMap& enginePar
 }
 #endif
 
+const Variant& Engine::GetParameter(const ea::string& name) const
+{
+    const auto iter = parameters_.find(name);
+    if (iter != parameters_.end() && !iter->second.IsEmpty())
+        return iter->second;
+    const auto descIter = parameterDesc_.find(name);
+    return descIter != parameterDesc_.end() ? descIter->second.defaultValue_ : Variant::EMPTY;
+}
+
 const Variant& Engine::GetParameter(const StringVariantMap& parameters, const ea::string& name, const Variant& defaultValue)
 {
     const auto iter = parameters.find(name);
     return iter != parameters.end() ? iter->second : defaultValue;
+}
+
+void Engine::LoadConfigFiles()
+{
+    const auto configName = GetParameter(EP_CONFIG_NAME).GetString();
+    if (configName.empty())
+        return;
 
+    const auto* vfs = GetSubsystem<VirtualFileSystem>();
+
+    // Populate default values in config file
+    ConfigFile mergedConfig(context_);
+    for (auto& keyValue : parameterDesc_)
+    {
+        if (keyValue.second.configOverride_)
+        {
+            mergedConfig.SetDefaultValue(keyValue.first, keyValue.second.defaultValue_);
+            if (HasParameter(keyValue.first))
+                mergedConfig.SetValue(keyValue.first, GetParameter(keyValue.first));
+        }
+    }
+
+    // Merge with values from config files.
+    mergedConfig.MergeFile(configName);
+
+    // Set parameters from merged values.
+    for (auto& keyValue : mergedConfig.GetValues())
+    {
+        parameters_[keyValue.first] = keyValue.second;
+    }
+}
+
+void Engine::SaveConfigFile()
+{
+    auto configName = GetParameter(EP_CONFIG_NAME).GetString();
+    if (configName.empty())
+        return;
+
+    // Populate values in config file
+    ConfigFile config(context_);
+    for (auto& keyValue : parameterDesc_)
+    {
+        if (keyValue.second.configOverride_)
+        {
+            config.SetDefaultValue(keyValue.first, keyValue.second.defaultValue_);
+            config.SetValue(keyValue.first, GetParameter(keyValue.first));
+        }
+    }
+
+    config.SaveDiffFile(configName);
+}
+
+void Engine::PopulateDefaultParameters()
+{
+    parameterDesc_[EP_APPLICATION_NAME].Set(false, "Unspecified Application");
```

Comment:
It's minor issue, but can we use builder pattern here?
`parameterDesc_[EP_APPLICATION_NAME].Set("Unspecified Application")`
and
`parameterDesc_[EP_FRAME_LIMITER].Set(true).OverrideInConfig();`

Reply:
Sure, sounds good